packages/blob/src/index.js

```diff
@@ -70,3 +70,41 @@ export function isBlobURL( url ) {
 	}
 	return url.indexOf( 'blob:' ) === 0;
 }
+
+/**
+ * Downloads a file, e.g., a text or readable stream, in the browser.
+ * Appropriate for downloading smaller file sizes, e.g., < 5 MB.
+ *
+ * Example usage:
+ *
+ * ```js
+ * 	const fileContent = JSON.stringify(
+ * 		{
+ * 			"title": "My Post",
+ * 		},
+ * 		null,
+ * 		2
+ * 	);
+ * 	const fileName = 'file.json';
+ *
+ * 	downloadBlob( 'file.json', fileContent, 'application/json' );
+ * ```
+ *
+ * @param {string}   fileName    File Name.
+ * @param {BlobPart} content     File Content (BufferSource | Blob | string).
+ * @param {string}   contentType File mime type.
+ * @return {HTMLElement}        The created anchor element.
+ */
+export function downloadBlob( fileName = 'wp-download', content, contentType ) {
```

Comment:
Nit: I think `filename` as one word is more common.

The fact that it's optional isn't reflected in the docs. I'm personally not sure about making the first param optional while the second and third are required. Or maybe all the params should be optional?

Reply:
Thanks for the comments. I think they should be all required actually. It seems unintuitive to have the first arg optional, I agree. 

Giving `fileName` a default value was an afterthought that sort of stuck around.

Reply:
It was also my comment I referred to in https://github.com/WordPress/gutenberg/pull/56024#pullrequestreview-1724138878 but somehow it didn't get saved.